python/ray/experimental/workflow/workflow_manager.py

```diff
@@ -66,43 +60,34 @@ def resolve_step_inputs(args: List[RRef], kwargs: Dict[str, RRef]
     flexibility.
 
     Args:
-        args: List of workflow step input arguments.
-        kwargs: Dict of workflow step input arguments.
-
+        step_inputs: Workflow step inputs.
     Returns:
         Instances of arguments and resolved object refs.
     """
 
     resolved_object_refs = []
-    _args = []
-    _kwargs = {}
-    for a in args:
-        if isinstance(a, RRef):
-            obj, ref = resolve_object_ref(a)
-            _args.append(obj)
-            resolved_object_refs.append(ref)
-        else:
-            _args.append(a)
-    for k, v in kwargs:
-        if isinstance(v, RRef):
-            obj, ref = resolve_object_ref(v)
-            _kwargs[k] = obj
-            resolved_object_refs.append(ref)
-        else:
-            _kwargs[k] = v
+    objects_mapping = []
+    inputs, workflow_outputs, object_refs = step_inputs
+    for rref in workflow_outputs:
+        obj, ref = resolve_object_ref(rref)
+        objects_mapping.append(obj)
+        resolved_object_refs.append(ref)
+    with serialization_context.workflow_args_resolving_context(
+            objects_mapping, object_refs):
+        _args, _kwargs = _deref_arguments(inputs)
     return _args, _kwargs, resolved_object_refs
 
 
 class WorkflowStepFunction:
```

Comment:
I'm thinking about what should we put into step function because right now, step function looks like coupled with workflow. For example, workflow have reference of object in step function and step function has reference to workflow.

Could it be better if we make `step` function a creator of Workflow only?

Reply:
I have made changes to decouple them.